### PR TITLE
[PAR-4291] display pp offer on product list view

### PR DESCRIPTION
**Pull request by jay-faust-extend**
# Description

Added the functionality to open the pp offer modal upon clicking `Add to Cart` button from the product list page. Tested with guidance toggled and visually there was no difference, the only difference was the plan doesn't get added to the cart with our module enabled (coming in [PAR-4187](https://helloextend.atlassian.net/browse/PAR-4187)). I also verified that the modal doesn't render if pp isn't enabled in our module.

## Ticket

https://helloextend.atlassian.net/browse/PAR-4262

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactor (new code changes but everything functions the same)

## Checklist

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have deployed and validated my code on a sandbox
- [x] I have added tests that prove my fix is effective or that my feature works

[PAR-4187]: https://helloextend.atlassian.net/browse/PAR-4187?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: view/frontend/web/js/view/catalog/product/product-protection-modal-offer.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+define(['jquery', 'extendSdk', 'ExtendMagento'], function ($, Extend, ExtendMagento) {
+  'use strict'
+
+  return function (config, element) {
+    Extend.config({ storeId: config[0].extendStoreUuid, environment: config[0].activeEnvironment })
+
+    const productSku = config[0].productSku
+
+    const addToCartForm = $('#product_protection_modal_offer_' + productSku)
```
Comment (jay-faust-extend) [inline view/frontend/web/js/view/catalog/product/product-protection-modal-offer.js]: I tried using vanilla js here but I couldn't find the initial div that way, not sure if jQuery waits for the document to be ready behind the scenes or something. If we care enough about avoiding using jQuery I'm open to suggested vanilla js solutions.

File: view/frontend/templates/catalog/product/view/product-protection-modal-offer.phtml [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+                    "extendStoreUuid": "<?= $extendStoreUuid ?>",
+                    "activeEnvironment": "<?= $activeEnvironment ?>",
+                    "productSku": "<?= $productSku ?>",
+                    "productPrice": "<?= $productPrice ?>",
+                    "productCategory": "<?= $categoryName ?>"
+                }
+            ]
+        }
+    }
+</script>
```
Comment (scott-taylor-extend) [inline view/frontend/templates/catalog/product/view/product-protection-modal-offer.phtml]: Can you add a new line here at the end of the file?

**Review by milesmcleod — DISMISSED** [review]

File: view/frontend/web/js/view/catalog/product/product-protection-modal-offer.js (hunk 1 above)
Comment (milesmcleod) [reply view/frontend/web/js/view/catalog/product/product-protection-modal-offer.js]: i think this is fine for now. my suggestion would be
```javascript
document.querySelector('#product_protection_modal_offer_' + productSku)
?.closest('.product.actions.product-item-actions')
?.querySelector('.actions-primary')
?.querySelector('form[data-role="tocart-form"]')
```
i wasn't aware of any differences between `$()` and `.querySelector` in regards to timing with dom content loading. i think this is fine but we might want a tech debt ticket to replace this in the future if we can find a simple vanilla way to make it work.

Comment (scott-taylor-extend) [reply view/frontend/web/js/view/catalog/product/product-protection-modal-offer.js]: @jay-faust-extend I'd go with the alternative above suggested by Miles.

Comment (jay-faust-extend) [reply view/frontend/web/js/view/catalog/product/product-protection-modal-offer.js]: @scott-taylor-extend I tried the above with no success, I'll try again

**Review by scott-taylor-extend — APPROVED** [review]

**Review by milesmcleod — APPROVED** [review]

**Pull request merged**